### DISCUSSION TITLE
Additional JMeter command line  args

### DISCUSSION THREAD

**Discussion by javid90khan** [qualersoft/jmeter-gradle-plugin#39] [Ideas]
Hi!

I need the ability to pass on Proxy Config with '-H' and '-P' proxy params. I have tried sysProp but that doesn't really get picked up by the http samplers. Is there a way to pass it already?

As a workaround, I Just added a HTTP defaults element and passing -J param for config. Serves my purpose for now.

Thanks

**Comment by mathze** [comment]
Hi @javid90khan,

`--J` is the right option to do what you want (for now) as this values are made available through jmeters "property resolving system". 
`--sysProp` has a different meaning as they are meant to configure jmeter and the libraries it uses. E.g. to set the secret of the keypass-file for certificate lookup etc. They are not provided through jmeters "property resolution system" (as far as I understood it). I also run in this "issue" and it took me some time to figure this out, as it is not documented thaaat explicit. :)

But back to your original request. Could you please give me a little more insights in the scenario you try to solve. Especially why this requires the options to being configurable through the plugin.

Many thanks in advantage

**Reply by javid90khan** [reply]
My original scenario was to run the tests from different machines. Based on where they are in the network, some need proxy and some don’t. 

I am happy with the workaround for now but perhaps consider a way to pass any raw JMeter command line arg.

**Reply by mathze** [reply]
Ah ok, I see the case.
Just to understand the process better. So when you start your jmeter gradle task on a client, You check in gradle (by what ever) if you execute on a client that has a proxy and iff you set the required properties. Or how does it work?
Btw. does the proxy settings are the same on each client or do they differ between clients?

**Reply by javid90khan** [reply]
Imagine this. You run a test plan that invokes publicly open (internet) endpoints. When you run it from your local machine, you can reach the internet without a proxy.
When u run the same on a network restricted build server, where internet traffic is only allowed through proxy and the server doesn’t have it configured on system level.

I now have -J variables that I can easily toggle per environment, so it’s all good.

**Reply by mathze** [reply]
I read a bit further in the [doc](https://jmeter.apache.org/usermanual/get-started.html#proxy_server) and read the following:
> If the proxy host and port are provided, then JMeter sets the following System properties:
>  * http.proxyHost
> ....

So there are system properties within jmeter for this purpose.
And since you said "I have tried sysProp" I'm now uncertain if there is a bug in the plugin.
Could you please provide me with an example cmd call you've used.

Many thanks in advanced
